my_class.cpp

```diff
@@ -0,0 +1,165 @@
+#include "my_class.h"
+#include <iostream>
+
+using std::cin;
+using std::cout;
+using std::endl;
+using std::string;
+
+void student::resizeGrades()
+{
+    float* newGrades = new float [currentCapacity*2];
+    for (int i = 0; i < currentCapacity; i++)
+    {
+        newGrades[i] = grades[i];
+    }
+    delete[] grades;
+    currentCapacity *= 2;
+    grades = newGrades;
+    newGrades = nullptr;
+}
+
+student::student()
+{
+    cout << "What is student's number in class?" << endl;
+    cin >> number;
+    cout << "What's the student's name?" << endl;
+    cin.ignore();
+    getline(cin, name);
+    numberGrades = 0;
+}
+
+int student::getNumber()
+{
+    return number;
+}
+
+string student::getName()
+{
+    return name;
+}
+
+void student::setNumber(int num)
+{
+    number = num;
+}
+
+void student::setName(string newName)
+{
+    name = newName;
+}
+    
+int student::getNumberGrades()
+{
+    return numberGrades;
+}
+
+void student::addNewGrade(float grade)
+{
+    if(numberGrades == MAX_GRADES)
+    {
+        cout << "This student has enough grades. " << endl;
+        return;
+    }
+    if(currentCapacity == numberGrades)
+    {
+        resizeGrades();
+    }
+    grades[numberGrades] = grade;
+    sum += grade;
+    numberGrades += 1;
+}
+
+bool student::areGrades()
```

Comment:
What is this function supposed to return? If the student has any grades at all?
If yes, renaming suggestion: areGradesReceived() to which the answer would be yes or no, true or false.

Reply:
Yeah, the purpose is to show if the student has any grades at all. Thank you for the suggestion.